addOns/ascanrules/src/test/java/org/zaproxy/zap/extension/ascanrules/ExternalRedirectScanRuleUnitTest.java

```diff
@@ -178,6 +241,34 @@ void shouldReportRedirectWithLocationOrRefreshHeaderIfSchemeIsAdded() throws Exc
         assertThat(alertsRaised.get(0).getAttack().startsWith(HttpHeader.HTTP), equalTo(false));
     }
 
+    @Test
+    void shouldNotReportRedirectWithLocationHeaderIfPayloadIsInParam() throws Exception {
```

Comment:
Well I want to ensure we're testing both as they were reported separately and it keeps us from letting something break in the future. However it could still be parameterised and so could the handler.

Reply:
I didn't say to stop testing any of the cases.